pragma/tests/fetchers_test.py

```diff
@@ -17,12 +22,50 @@
     FUTURE_FETCHER_CONFIGS,
     ONCHAIN_FETCHER_CONFIGS,
 )
+from pragma.tests.fixtures.devnet import get_available_port, get_compiler_manifest
 
 PUBLISHER_NAME = "TEST_PUBLISHER"
 
 # %% SPOT
 
 
+@pytest.fixture(scope="module")
+def forked_client(module_mocker, pytestconfig) -> Client:
+    """
+    This module-scope fixture prepares a forked starknet-dev
+    client for e2e testing.
+
+    :return: a starknet Client
+    """
+    net = pytestconfig.getoption("--net")
+    port = get_available_port()
+    command = [
+        "poetry",
+        "run",
+        "starknet-devnet",
+        "--fork-network",
+        "alpha-mainnet",
+        "--host",
+        "127.0.0.1",
+        "--port",
+        str(port),
+        "--accounts",  # deploys specified number of accounts
+        str(1),
+        "--seed",  # generates same accounts each time
+        str(1),
+        *get_compiler_manifest(),
+    ]
+    subprocess.Popen(command)
+    time.sleep(10)
+    pragma_client = PragmaClient(net, port=port)
+    module_mocker.patch.object(
+        AvnuFetcher,
+        "_pragma_client",
+        return_value=pragma_client,
+    )
```

Comment:
Tbh, not sure if this is the best place to put this. At the very least, this fixture ensures that the pragma client is only patched once for this module and is "cleaner" by not being part of the actual testing code.